src/contracts/signing-key-events-cache/signing-key-events-cache.service.ts

```diff
@@ -70,12 +76,18 @@ export class SigningKeyEventsCacheService {
       this.getCachedEvents(),
     ]);
 
+    // TODO: do we need to check cache validity ?
```

Comment:
`const isCacheValid = this.validateCacheBlock(cachedEvents, blockNumber);`
The same applies to the deposit events cache.

In the `initialize` function, we check `validateCacheBlock`, but in `handleNewBlock`, we don't.

However, since the `initialize` function is executed before `handleNewBlock`, we can be confident that the cache is valid during `handleNewBlock` execution. Therefore, we don't need to validate the cache again in `handleNewBlock`.

However, the `toBlock` value could be less than `firstNotCachedBlock`. We might need to implement an early exit in this case, and do the same for deposit events.